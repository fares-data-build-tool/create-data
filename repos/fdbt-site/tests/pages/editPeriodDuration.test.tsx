import { shallow } from 'enzyme';
import React from 'react';
import EditPeriodDuration from '../../src/pages/editPeriodDuration';

describe('pages', () => {
    describe('editPeriodDuration', () => {
        it('should render editPeriodDuration page correctly', () => {
            const wrapper = shallow(
                <EditPeriodDuration
                    errors={[]}
                    csrfToken=""
                    productDurationValue="1"
                    productDurationUnit="week"
                    backHref=""
                />,
            );
            expect(wrapper).toMatchSnapshot();
        });
        it('should render editPeriodDuration page correctly with an error', () => {
            const wrapper = shallow(
                <EditPeriodDuration
                    errors={[{ id: 'edit-period-duration-quantity', errorMessage: 'Product duration cannot be empty' }]}
                    csrfToken=""
                    productDurationValue=""
                    productDurationUnit="week"
                    backHref=""
                />,
            );
            expect(wrapper).toMatchSnapshot();
        });
    });
});
