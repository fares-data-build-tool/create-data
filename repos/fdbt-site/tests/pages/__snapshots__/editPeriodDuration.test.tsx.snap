// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages editPeriodDuration should render editPeriodDuration page correctly 1`] = `
<TwoThirdsLayout
  description="Edit Period Duration page of the Create Fares Data Service"
  errors={Array []}
  title="Edit Period Duration - Create Fares Data Service"
>
  <CsrfForm
    action="/api/editPeriodDuration"
    csrfToken=""
    method="post"
  >
    <div
      className="govuk-form-group"
    >
      <ErrorSummary
        errors={Array []}
      />
      <h1
        className="govuk-heading-l"
        id="edit-period-duration-page-heading"
      >
        Edit Period Duration
      </h1>
      <label
        className="govuk-label"
        htmlFor="edit-period-duration"
      >
        Period duration
      </label>
      <span
        className="govuk-hint"
        id="edit-period-duration-hint"
      >
        For example, 3 days
      </span>
      <ExpirySelector
        carnet={false}
        defaultDuration="1"
        defaultUnit="week"
        errors={Array []}
        hideFormGroupError={true}
        hintId="edit-period-duration-hint"
        quantityId="edit-period-duration-quantity"
        quantityName="periodDurationInput"
        school={false}
        unitId="edit-period-duration-unit"
        unitName="periodDurationUnitsInput"
      />
      <br />
      <input
        className="govuk-button"
        id="continue-button"
        type="submit"
        value="Continue"
      />
    </div>
  </CsrfForm>
</TwoThirdsLayout>
`;

exports[`pages editPeriodDuration should render editPeriodDuration page correctly with an error 1`] = `
<TwoThirdsLayout
  description="Edit Period Duration page of the Create Fares Data Service"
  errors={
    Array [
      Object {
        "errorMessage": "Product duration cannot be empty",
        "id": "edit-period-duration-quantity",
      },
    ]
  }
  title="Edit Period Duration - Create Fares Data Service"
>
  <CsrfForm
    action="/api/editPeriodDuration"
    csrfToken=""
    method="post"
  >
    <div
      className="govuk-form-group"
    >
      <ErrorSummary
        errors={
          Array [
            Object {
              "errorMessage": "Product duration cannot be empty",
              "id": "edit-period-duration-quantity",
            },
          ]
        }
      />
      <h1
        className="govuk-heading-l"
        id="edit-period-duration-page-heading"
      >
        Edit Period Duration
      </h1>
      <label
        className="govuk-label"
        htmlFor="edit-period-duration"
      >
        Period duration
      </label>
      <span
        className="govuk-hint"
        id="edit-period-duration-hint"
      >
        For example, 3 days
      </span>
      <ExpirySelector
        carnet={false}
        defaultUnit="week"
        errors={
          Array [
            Object {
              "errorMessage": "Product duration cannot be empty",
              "id": "edit-period-duration-quantity",
            },
          ]
        }
        hideFormGroupError={true}
        hintId="edit-period-duration-hint"
        quantityId="edit-period-duration-quantity"
        quantityName="periodDurationInput"
        school={false}
        unitId="edit-period-duration-unit"
        unitName="periodDurationUnitsInput"
      />
      <br />
      <input
        className="govuk-button"
        id="continue-button"
        type="submit"
        value="Continue"
      />
    </div>
  </CsrfForm>
</TwoThirdsLayout>
`;
